framework/generated/dx12_generators/dx12_wrapper_body_generator.py

```diff
@@ -772,6 +783,49 @@ def write_class_member_def(self, class_info, indent=''):
                     encode_args += wrapped_args
             expr += encode_args + ');\n'
 
+            if is_override is False and 'ID3D12GraphicsCommandList' in class_name:
+                indent1 = self.increment_indent(indent)
+                indent2 = self.increment_indent(indent1)
+                indent3 = self.increment_indent(indent2)
+                indent4 = self.increment_indent(indent3)
+                expr += '\n'
+                expr += indent + 'if(manager->GetTrimBoundary() == CaptureSettings::TrimBoundary::kDrawCalls)\n'
+                expr += indent + '{\n'
+                expr += indent1 + 'manager->DecrementCallScope();\n'
+                expr += indent1 + 'auto trim_draw_calls_command_sets = manager->GetCommandListsForTrimDrawCalls(this, format::ApiCall_{}_{});\n'.format(class_name, method_name)
+                expr += indent1 + 'for(auto& command_set : trim_draw_calls_command_sets)\n'
+                expr += indent1 + '{\n'
+
+                if class_name != 'ID3D12GraphicsCommandList':
+                    expr += indent2 + 'graphics::dx12::{}ComPtr command_list = nullptr;\n'.format(class_name)
+                    expr += indent2 + 'command_set.list->QueryInterface(IID_PPV_ARGS(&command_list));\n'
+                    expr += indent2 + 'auto* wrapper = reinterpret_cast<{}_Wrapper*>(command_list.GetInterfacePtr());\n'.format(class_name)
```

Comment:
I think these lines could be changed to avoid the QueryInterface call. Since the call scope is decremented to 0 here, QueryInterface may have some side-effects for capture/tracking and is probably cleaner to avoid using it.

```
  expr += indent2 + 'auto* base_wrapper = reinterpret_cast<ID3D12GraphicsCommandList_Wrapper*>(command_set.list.GetInterfacePtr());\n'
  expr += indent2 + 'auto* wrapper = dynamic_cast<{}_Wrapper*>(base_wrapper);\n'.format(class_name)
  expr += indent2 + 'GFXRECON_ASSERT(wrapper != nullptr);\n'
```

Which results in this code in `generated_dx12_wrappers.cpp`:
```
  auto* base_wrapper = reinterpret_cast<ID3D12GraphicsCommandList_Wrapper*>(command_set.list.GetInterfacePtr());
  auto* wrapper = dynamic_cast<ID3D12GraphicsCommandList1_Wrapper*>(base_wrapper);
  GFXRECON_ASSERT(wrapper != nullptr);
```

Reply:
One change to the code above. Instead of `dynamic_cast` I think `static_cast` can be used since we know all `*CommandList_Wrappers` are created as the most derived class `ID3D12GraphicsCommandList9_Wrapper` in `void WrapID3D12CommandList(REFIID riid, void** object, DxWrapperResources* resources)` and we can drop the assert. So instead, the generated code would be:

```
  auto* base_wrapper = reinterpret_cast<ID3D12GraphicsCommandList_Wrapper*>(command_set.list.GetInterfacePtr());
  auto* wrapper = static_cast<ID3D12GraphicsCommandList1_Wrapper*>(base_wrapper);
```

Reply:
How about just `auto* wrapper = reinterpret_cast<ID3D12GraphicsCommandList1_Wrapper*>(command_set.list.GetInterfacePtr());`  to reduce one cast. Is it safe?

Reply:
fixed